### ISSUE TITLE
acrn-on-qemu-SMAP-problem

### ISSUE THREAD

**Issue by Simpronic** [projectacrn/acrn-hypervisor#8571]
Hello, I should run ACRN on QEMU.
I successfully compiled all acrn's sources and installed them.
I can find the kenel image in /boot folder.
But i have a problem in booting arcn and it gives me this message: 

error: Can't get controller info..
Non-UEFI environment detected.
Multiboot Bootloader: GRUB 2.04-1ubuntu26.17
detect_hardware_support, SMAP not supported
[0us][cpu=0][(null)][serv=1][seq=1]:PANIC: init_pcpu_pre line 153

[0us][cpu=0][(null)][serv=1][seq=1]:hardware not support!

How can i do ? 

I controlled the features of my cpu, but it doesn't seem to support SMAP, so i can't boot ACRN ?

**Comment by Simpronic** [comment]
UPDATE: i found in the kernel that i can deactivate SMAP and i deactivated it
But there has been another problem with CLFLUSHOPT, OPcode not in my ISA (i have only CLFLUSH).
I changed all the instruction with CLFLUSHOPT with CLFLUSH but when i booted the kenel i ended up with this: 


Loading ACRN hypervisor ...
error: Can't get controller info..
Non-UEFI environment detected.
Multiboot Bootloader: GRUB 2.04-1ubuntu26.17
ACRN Hypervisor
[3326137us][cpu=0][(null)][sev=2][seq=1]:HV version 3.0-2024-03-27 13:15:36-506c87cfa-dirty DBG (daily tag:acrn-2022w27.1-180000p) shared@qemu build by marco, start time 3315718us
[3328095us][cpu=0][(null)][sev=2][seq=2]:API version 1.0
[3328699us][cpu=0][(null)][sev=2][seq=3]:Detect processor: Intel Atom Processor (Denverton)
[3329738us][cpu=0][(null)][sev=1][seq=4]:SECURITY WARNING!!!!!!
[3330427us][cpu=0][(null)][sev=1][seq=5]:Please apply the latest CPU uCode patch!
[3333058us][cpu=0][(null)][sev=2][seq=6]:Assertion failed in file acpi_parser/dmar_parse.c,line 120 : fatal error
[3334229us][cpu=0][(null)][sev=2][seq=7]:
 delta = (actual_load_address - CONFIG_HV_RAM_START) = 0x0

[3335265us][cpu=0][(null)][sev=2][seq=8]:
Host Stack: CPU_ID = 0

[3335963us][cpu=0][(null)][sev=2][seq=9]:addr(0x1120bb90)	0x0000000000000000  0x0000000000000013  0x0000000011247f00  0x000000007ffe2b9b

[3337349us][cpu=0][(null)][sev=2][seq=10]:addr(0x1120bbb0)	0x000000001120bc20  0x0000000011012213  0x0000000011055880  0x000000007ffe2c43

[3338733us][cpu=0][(null)][sev=2][seq=11]:addr(0x1120bbd0)	0xffffffff1120bc10  0x0000000011247f80  0x00000000000000fe  0x0000000000000000

[3340151us][cpu=0][(null)][sev=2][seq=12]:addr(0x1120bbf0)	0x00000000000000fc  0x0000000000000000  0x0000000000000000  0x0000000000000000

[3341777us][cpu=0][(null)][sev=2][seq=13]:addr(0x1120bc10)	0x0000000000000000  0x0000000000000000  0x000000001120bc30  0x000000001100d800

[3343522us][cpu=0][(null)][sev=2][seq=14]:addr(0x1120bc30)	0x000000001120bc60  0x0000000011009499  0x0000000000000000  0x0000000000000000

[3345250us][cpu=0][(null)][sev=2][seq=15]:addr(0x1120bc50)	0x0000000000000000  0x0000000000000000  0x000000001120bc80  0x00000000110081e9

[3346503us][cpu=0][(null)][sev=2][seq=16]:addr(0x1120bc70)	0x00000000110010e2  0x0000000000000000  0x0000000011062310  0x00000000696e746c

[3347756us][cpu=0][(null)][sev=2][seq=17]:addr(0x1120bc90)	0x0000000000000000  0x0000000000000000  0x0000000100000000  0x0000000000000000

[3349024us][cpu=0][(null)][sev=2][seq=18]:addr(0x1120bcb0)	0x0000000000000000  0x0000000000000000  0x0000000000000000  0x0000000000000000

[3350635us][cpu=0][(null)][sev=2][seq=19]:addr(0x1120bcd0)	0x0000000000000000  0x0000000000000000  0x0000000000000000  0x0000000000000000

[3352444us][cpu=0][(null)][sev=2][seq=20]:addr(0x1120bcf0)	0x0000000000000000  0x0000000000000000  0x0000000000000000  0x0000000000000000

[3354873us][cpu=0][(null)][sev=2][seq=21]:addr(0x1120bd10)	0x0000000000000000  0x0000000000000000  0x0000000000000000  0x0000000000000000

[3356712us][cpu=0][(null)][sev=2][seq=22]:addr(0x1120bd30)	0x0000000000000000  0x0000000000000000  0x0000000000000000  0x0000000000000000

[3358437us][cpu=0][(null)][sev=2][seq=23]:addr(0x1120bd50)	0x0000000000000000  0x0000000000000000  0x0000000000000000  0x0000000000000000

[3360262us][cpu=0][(null)][sev=2][seq=24]:addr(0x1120bd70)	0x0000000000000000  0x0000000000000000  0x0000000000000000  0x0000000000000000

[3362053us][cpu=0][(null)][sev=2][seq=25]:

[3362649us][cpu=0][(null)][sev=2][seq=26]:Host Call Trace:

[3363464us][cpu=0][(null)][sev=2][seq=27]:----> 0x0000000011012213

[3364385us][cpu=0][(null)][sev=2][seq=28]:----> 0x000000001100d800

[3365308us][cpu=0][(null)][sev=2][seq=29]:----> 0x0000000011009499

[3366222us][cpu=0][(null)][sev=2][seq=30]:----> 0x00000000110081e9

[3367133us][cpu=0][(null)][sev=2][seq=31]:
 
and then it freezes

**Comment by MingzhenHan** [comment]
I also met the same problem with you, have you solved ?

**Comment by Simpronic** [comment]
No, I commented the assertion line and recompiled the hypervisor code.
When I enter into the grub I choose ACRN hypervisor and it enters in a sort of boot loop returning always in the selection section

**Comment by MingzhenHan** [comment]
[3330427us][cpu=0][(null)][sev=1][seq=5]:Please apply the latest CPU uCode patch!
What do you think of this msg？
Should we apply the latest CPU microcode patch on guest or host by command
`sudo apt install intel-microcode`

**Comment by Simpronic** [comment]
I didn't tried it, so I don't know if it can solve the problems.
If you try, let me know if it does something

**Comment by MingzhenHan** [comment]
Pity, it doesn't work. What 's your host ubuntu version and Qemu emulator version?

**Comment by Simpronic** [comment]
My host Ubuntu version is 22.04 LTS and for the ServiceVM I use 20.04 (the one that the official guide use) .
Qemu version is 6.0 (I tried also the latest by compiling the binaries, but that version gave me some problems)

**Comment by Simpronic** [comment]
I don't know if it matters.
I was thinking about some features that my CPU doesn't have.
Apart from SMAP, my CPU missed also CLFLUSHOPT.
And in my opinion also that assertion is important, but I don't know how to solve or debug this problem
